src/box/txn_limbo.c

```diff
@@ -176,7 +176,8 @@ txn_limbo_append(struct txn_limbo *limbo, uint32_t id, struct txn *txn)
 }
 
 static inline void
-txn_limbo_remove(struct txn_limbo *limbo, struct txn_limbo_entry *entry)
+txn_limbo_pop_confirmed_entry(struct txn_limbo *limbo,
+			      struct txn_limbo_entry *entry)
 {
 	assert(!rlist_empty(&entry->in_queue));
 	assert(txn_limbo_first_entry(limbo) == entry);
```

Comment:
Please, add `assert(entry->is_commit)`.

Reply:
Done.

Reply:
Do we really wanna add a function argument just to pass it to the assertion? I don't like that a lot

Reply:
Since entry doesn't always has `is_commit` flag, let's not add so sophisticated assertion.

Reply:
Yes, the new assertion looks indeed quite overlycomplex, and it reveals a bug. See my comment below.